src/hotspot/share/compiler/compileBroker.cpp

```diff
@@ -1023,6 +1023,20 @@ void CompileBroker::init_compiler_threads() {
 
 void CompileBroker::possibly_add_compiler_threads(JavaThread* THREAD) {
 
+  int old_c2_count = 0, new_c2_count = 0, old_c1_count = 0, new_c1_count = 0;
+  const int c2_tasks_per_thread = 2, c1_tasks_per_thread = 4;
```

Comment:
Any reason to have such numbers (2 and 4)?  Any experiments were done to select the best numbers?

Reply:
Please note that these constants are not new. I have only given them names. I had done some experiments when implementing [JDK-8198756](https://bugs.openjdk.org/browse/JDK-8198756) for JDK11. C1 is faster than C2. Therefore, we can have more C1 tasks per C1 thread.

Reply:
Good.